internal/mode/static/manager.go

```diff
@@ -376,8 +377,18 @@ func setInitialConfig(
 	defer cancel()
 
 	var config ngfAPI.NginxGateway
-	if err := reader.Get(ctx, configName, &config); err != nil {
-		return err
+	if err := wait.PollUntilContextCancel(
+		ctx,
+		500*time.Millisecond,
+		true, /* poll immediately */
+		func(ctx context.Context) (bool, error) {
+			if err := reader.Get(ctx, configName, &config); err != nil {
+				return false, nil //nolint:nilerr // returning an error cancels the polling
```

Comment:
what if an error is not recoverable here? do we need to check the  error type/value?

Reply:
Good call, I should be able to check the error type